src/core/transaction.js

```diff
@@ -435,6 +435,15 @@ Transaction.prototype.signingHash = function(testnet) {
   return this.hash(testnet ? 'HASH_TX_SIGN_TESTNET' : 'HASH_TX_SIGN');
 };
 
+Transaction.prototype.signingData = function() {
+  return this._signingData('HASH_TX_SIGN').buffer;
+};
+
+Transaction.prototype._signingData = function(prefix) {
+  return SerializedObject.from_json(
+        this.tx_json, hashprefixes[prefix].toString(16));
```

Comment:
I guess theoretically there could be a problem if the hash prefix started with a 0.